### DISCUSSION TITLE
ERROR(101): Problem with MATL

### DISCUSSION THREAD

**Discussion by DOGzym** [firemodels/fds#12775] [Q&A] [answered]
Hi all,
Today I updated FDS to version 6.9.1(The version used in the past was 6.7.9). But the same code cannot run in the 6.9.1 new version. Program failed: ERROR (101): Problem with MATL number1, line number 49(Here is exactly my code to set up the solid pyrolysis model). I want to know why the same code works fine in the old version but not in the new version. Is this due to some changes in the 6.9.1 new version?
The code of this part is as follows. 
&MATL ID = 'PMMA'
      SPECIFIC_HEAT_RAMP = 'PMMA_SPECHEAT_FUNCTION'
      CONDUCTIVITY_RAMP  = 'PMMA_CONDUCT_FUNCTION'
      REFRACTIVE_INDEX = 1.576
      DENSITY = 1160
      ABSORPTION_COEFFICIENT = 2250
      EMISSIVITY = 0.95
      A(1) = 8.60E12
      E(1) = 188000
      SPEC_ID = 'MMA'
      N_REACTIONS = 1
      NU_SPEC = 1.0
      HEAT_OF_REACTION = 846 /

&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 27    F = 1.7010 /
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 37    F = 1.7377 /
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 47    F = 1.8111 /
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 57    F = 1.8478/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 67    F = 1.8845/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 77    F = 1.9212/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 87    F = 1.9579/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 97    F = 1.9946/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 107   F = 2.0313/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 117   F = 2.0680/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 127   F = 2.1047/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 137   F = 2.1414/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 147   F = 2.1781/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 157   F = 2.2148/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 167   F = 2.2515/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 177   F = 2.2882/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 187   F = 2.3249/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 197   F = 2.3616/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 207   F = 2.3983/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 217   F = 2.4350/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 227   F = 2.4717/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 237   F = 2.5084/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 247   F = 2.5451/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 257   F = 2.5818/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 267   F = 2.6185/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 277   F = 2.6552/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 287   F = 2.6919/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 297   F = 2.7286/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 307   F = 2.7653/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 317   F = 2.8020/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 327   F = 2.8387/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 337   F = 2.8754/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 347   F = 2.9121/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 357   F = 2.9488/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 367   F = 2.9855/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 377   F = 3.0222/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 387   F = 3.0589/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 397   F = 3.0956/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 407   F = 3.1323/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 417   F = 3.1690/
&RAMP ID = 'PMMA_SPECHEAT_FUNCTION'  T = 427   F = 3.2057/
   
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 27    F = 0.3360/	
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 37    F = 0.3322/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 47    F = 0.3284/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 57    F = 0.3246/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 67    F = 0.3208/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 77    F = 0.3170/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 87    F = 0.3132/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 97    F = 0.3094/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 107   F = 0.1788/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 117   F = 0.1764/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 127   F = 0.1740/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 137   F = 0.1716/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 147   F = 0.1692/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 157   F = 0.1668/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 167   F = 0.1644/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 177   F = 0.1620/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 187   F = 0.1596/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 197   F = 0.1572/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 207   F = 0.1548/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 217   F = 0.1524/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 227   F = 0.1500/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 237   F = 0.1476/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 247   F = 0.1452/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 257   F = 0.1428/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 267   F = 0.1404/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 277   F = 0.1380/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 287   F = 0.1356/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 297   F = 0.1332/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 307   F = 0.1308/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 317   F = 0.1284/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 327   F = 0.1260/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 337   F = 0.1236/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 347   F = 0.1212/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 357   F = 0.1188/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 367   F = 0.1164/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 377   F = 0.1140/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 387   F = 0.1116/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 397   F = 0.1092/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 407   F = 0.1068/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 417   F = 0.1044/
&RAMP ID = 'PMMA_CONDUCT_FUNCTION'  T = 427   F = 0.1020/

**Comment by drjfloyd** [comment]
The back of the User's Guide has a table of error codes that take you the section of the guide containing details to help you resolve the error.  Error code 101 will take you to the tables containing valid inputs for each namelist category. Here is where you can check you inputs against the inputs valid in the current version.

**Comment by rmcdermo** [comment]
Just to add a little to the general discussion of backward compatibility... When we issue a "minor" release, we may change input parameters.  The minor version number is the second part of the whole version number.  So, in FDS 6.9.0, this is major version 6, minor version 9, and maintenance version 0.  You should expect the same input file to work with a new maintenance release, but not necessarily with a new minor release (and with a major release, fuggettaboutit).

The strict backward compatibility of maintenance releases is somewhat of a new policy, and you can thank Thunderhead Engineering for advocating for this.  So, it is probably that this is not true of older maintenance releases.

**Comment by DOGzym** [comment]
Thanks for your suggestion!

I have checked the latest FDS User's Guide but could not find the reason why the new version cannot run the same code in the error code section. (MATL line)

The reason why I tried to use the new version of FDS is that I wanted to set the 'GAUGE HEAT FLUX GAS' gas phase heat flow probe point, but the 6.7.9 version I used before and the 6.8.0 version I just tried reported an error: DEVC QUANTITY GUAGE HEAT FLUX GAS not found.

I have tried using versions 6.9.1 and 6.9.0. The same code reports an error on the MATL line in both versions, so I'm not sure whether 'GAUGE HEAT FLUX GAS' can be set in these two versions.

**Comment by rmcdermo** [comment]
Using syntax highlighting is a good way to find these errors.  REFRACTIVE_INDEX is not a parameter in the latest code.

![Screenshot 2024-04-11 at 9 39 23 AM](https://github.com/firemodels/fds/assets/4418497/b93f9112-48e3-408b-afa1-4fc27d8eb031)

I maintain an up-to-date syntax highlighter for Sublime Text with FDS [here](https://github.com/rmcdermo/SublimeFDS).

**Reply by DOGzym** [reply]
Thanks a lot!

**Reply by rmcdermo** [reply]
Glad it helped.  I should mention, there are couple of other syntax highlighters for FDS on our [Third Party Tools](https://github.com/firemodels/fds/wiki/Third-Party-Tools#syntax-highlighting) wiki page.

**Reply by DOGzym** [reply]
After deleting REFRACTIVE_INDEX, I ran the program in the latest version of FDS, but I still got the ERROR: QUANTITY GUAGE HEAT FLUX GAS not found.
This shows that the cause of the error has nothing to do with the FDS version.

**Reply by rmcdermo** [reply, marked as answer]
How do you spell "gauge"?

**Reply by DOGzym** [reply]
It's actually a spelling problem.

**Reply by faiz165** [reply]
no
